### PR TITLE
Allow catch-all THDs to apply to all completions.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Prior to this catch-alls were explicitly ignored not to populate the IntelliSense with an excess amount of `TagHelper`s. To maintain backwards compatible expectations this commit reverts that expectation.
- Scoped an existing test to only test its primary scenario.
- Added a new test to verify that catch-alls accurately apply to all completions (including dynamically added ones).

#1219

### CODE REVIEW

File: test/Microsoft.VisualStudio.LanguageServices.Razor.Test/DefaultTagHelperCompletionServiceTest.cs [hunk 1]
```diff
@@ -183,13 +216,9 @@ public void GetElementCompletions_AllowedChildrenAreIgnoredWhenAtRoot()
...
             };
-            var expectedCompletions = ElementCompletionResult.Create(new Dictionary<string, HashSet<TagHelperDescriptor>>()
-            {
-                ["p"] = new HashSet<TagHelperDescriptor>(),
-                ["em"] = new HashSet<TagHelperDescriptor>(),
-            });
+            var expectedCompletions = ElementCompletionResult.Create(new Dictionary<string, HashSet<TagHelperDescriptor>>());
 
-            var existingCompletions = new[] { "p", "em" };
+            var existingCompletions = Enumerable.Empty<string>();
```
Comment (ajaybhargavb) [inline test/Microsoft.VisualStudio.LanguageServices.Razor.Test/DefaultTagHelperCompletionServiceTest.cs]: Why did this have to change?

File: src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperCompletionService.cs [hunk 2]
```diff
@@ -74,19 +76,35 @@ public override ElementCompletionResult GetElementCompletions(ElementCompletionC
...
+                {
+                    UpdateCompletions(completionTagNames, catchAllDescriptor);
+                }
+            }
+
             var result = ElementCompletionResult.Create(elementCompletions);
             return result;
+
+
+            void UpdateCompletions(string tagName, TagHelperDescriptor possibleDescriptor)
```
Comment (ajaybhargavb) [inline src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperCompletionService.cs]: 💯  PranavPoints

File: test/Microsoft.VisualStudio.LanguageServices.Razor.Test/DefaultTagHelperCompletionServiceTest.cs (hunk 1 above)
Comment (NTaylorMullen) [inline test/Microsoft.VisualStudio.LanguageServices.Razor.Test/DefaultTagHelperCompletionServiceTest.cs]: Scoped in the test. The existing completions didn't help validate the tests expectations; they ended up crossing over with the new test I added.

Comment (ajaybhargavb) [inline test/Microsoft.VisualStudio.LanguageServices.Razor.Test/DefaultTagHelperCompletionServiceTest.cs]: Got it

**Pull request merged**